pkg/kessel/errors/errors.go

```diff
@@ -0,0 +1,52 @@
+package errors
+
+import (
+	"fmt"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type Error struct {
+	Code    codes.Code
+	Message string
+	Cause   error
+}
+
+func (e *Error) Error() string {
+	if e.Cause != nil {
+		return fmt.Sprintf("%s: %v", e.Message, e.Cause)
+	}
+	return e.Message
+}
+
+func (e *Error) GRPCStatus() *status.Status {
+	return status.New(e.Code, e.Message)
+}
+
+func Wrap(err error, message string) error {
```

Comment:
Should we create [error sentinels](https://gobyexample.com/errors) instead of plain wrapped errors?